File: dotnet/DesignPatterns/BehavioralDesignPatterns/CommandDesignPattern/CommandDesignPattern.Example1/TextEditor.cs
namespace CommandDesignPattern.Example1;

/// <summary>
/// Receiver - The text editor that performs the actual operations
/// </summary>
public class TextEditor
{
    private string _content = string.Empty;

    public string Content => _content;

    public void AppendText(string text)
    {
        _content += text;
        Console.WriteLine($"Text added: '{text}'");
    }

    public void DeleteText(int length)
    {
        if (length > _content.Length)
            length = _content.Length;

        _content = _content.Substring(0, _content.Length - length);
        Console.WriteLine($"Deleted {length} characters");
    }

    public void DisplayContent()
    {
        Console.WriteLine($"\n=== Editor Content ===");
        Console.WriteLine($"'{_content}'");
        Console.WriteLine($"=====================\n");
    }
}

File: nodejs/DesignPatterns/BehavioralDesignPatterns/CommandDesignPattern/CommandDesignPattern.Example1/src/TextEditor.ts
/**
 * Receiver — The text editor that performs the actual operations.
 */
export class TextEditor {
  private _content = "";

  get content(): string {
    return this._content;
  }

  appendText(text: string): void {
    this._content += text;
    console.log(`Text added: '${text}'`);
  }

  deleteText(length: number): void {
    const actual = Math.min(length, this._content.length);
    this._content = this._content.slice(0, this._content.length - actual);
    console.log(`Deleted ${actual} characters`);
  }

  displayContent(): void {
    console.log(`\n=== Editor Content ===`);
    console.log(`'${this._content}'`);
    console.log(`=====================\n`);
  }
}
